albatross.ml

```diff
@@ -533,82 +552,127 @@ module Make (S : Tcpip.Stack.V4V6) = struct
                                "albatross received error reading from %a \
                                 querying %a %a: %a"
                                Fmt.(pair ~sep:(any ":") Ipaddr.pp int)
-                               t.remote Vmm_core.Name.pp name
+                               (config.server_ip, config.server_port)
+                               Vmm_core.Name.pp name
                                (Vmm_commands.pp ~verbose:false)
                                cmd TLS.pp_error e))
                   | Error err -> Lwt.return_error err)))
 
-  let init stack server ?(port = 1025) cert key =
+  let init stack (configuration : Configuration.t) =
     let open Lwt.Infix in
-    let t =
-      { stack; remote = (server, port); cert; key; policies = Vmm_trie.empty }
-    in
     let cmd = `Policy_cmd `Policy_info in
-    match gen_cert t cmd "." with
-    | Error s -> invalid_arg s
-    | Ok certificates -> (
-        raw_query t certificates cmd reply >|= function
+    let name_for_cert =
+      Vmm_core.Name.of_string configuration.name |> Result.get_ok
+    in
+    match
+      key_cert ~is_ca:false ~cmd configuration.private_key configuration.name
+        (X509.Certificate.subject configuration.certificate)
+    with
+    | Error s ->
+        Logs.warn (fun m ->
+            m "init: key_cert failed for %s: %s" configuration.name s);
+        let state =
+          {
+            errors = Some "Initialization error with key certificate";
```

Comment:
So this is a local error, and it would be great to have a bit more detail -- why not adding the `s` here?